src/orders/views/OrderDetails/OrderNormalDetails/index.tsx

```diff
@@ -114,11 +114,34 @@ export const OrderNormalDetails: React.FC<OrderNormalDetailsProps> = ({
   const [fulfillmentWarehouse, setFulfillmentWarehouse] = React.useState<
     Warehouse
   >(null);
+
   React.useEffect(() => {
-    // @TODO this is wip
-    // exact logic for determining default
-    // warehouse will be added in future PR
-    setFulfillmentWarehouse(warehouses?.[0]);
+    const warehousesAvailability = warehouses?.map(warehouse => {
+      let linesAvailable = 0;
+
+      if (!order?.lines) {
+        return undefined;
+      }
+      order.lines.forEach(line => {
+        if (
+          line?.variant?.stocks?.find(
+            stock => stock.warehouse.id === warehouse.id
+          )
+        ) {
+          linesAvailable += 1;
+        }
+      });
+
+      return {
+        warehouse,
+        linesAvailable
+      };
+    });
+
+    const defaultWarehouse = warehousesAvailability?.reduce((prev, curr) =>
+      curr.linesAvailable > prev.linesAvailable ? curr : prev
```

Comment:
Won't this fail if there are no `lines` in `order`? We'll be calling `reduce` on an array of `undefined`s

Reply:
It doesn't actually, optional chaining works here

Reply:
Optional chaining you're referring to (I think) checks if the array exists, not if its content is filled with undefineds, right?

Reply:
![image](https://user-images.githubusercontent.com/13994677/156353323-0785b3f2-50f6-4d45-ae5d-ce8c507e9d2f.png)

Reply:
You're right, it fails on new orders

Reply:
I think you need to refactor the types a bit because it should detect that the array could contain `undefined`s